pkg/providers/instance/instance.go

```diff
@@ -298,11 +304,16 @@ func (p *DefaultProvider) getLaunchTemplateConfigs(ctx context.Context, nodeClas
 	if err != nil {
 		return nil, fmt.Errorf("getting launch templates, %w", err)
 	}
-	requirements := scheduling.NewNodeSelectorRequirementsWithMinValues(nodeClaim.Spec.Requirements...)
-	requirements[corev1beta1.CapacityTypeLabelKey] = scheduling.NewRequirement(corev1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, capacityType)
+
+	// Pick 1 NodePool (instanceType + AZ) if mulitple ODCRs match it, because CreateFleet does not allow
```

Comment:
I wonder if there's some work that we can do ahead of time so that we don't even create the duplicate launch templates